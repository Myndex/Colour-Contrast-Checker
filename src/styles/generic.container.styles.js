import styled, { css } from 'styled-components';
import { minWidth } from './settings.breakpoints.styles';

export const Wrapper = styled.div`
  position: relative;
`;

export const Container = styled.div`
  position: relative;
  padding-right: 5vw;
  padding-left: 5vw;

  ${props => props.fullHeight && css`
    min-height: 100vh;
  `}

  ${minWidth('992', () => css`
    max-width: 1400px;
    margin: 0 auto;
    padding-right: 15px;
    padding-left: 15px;
  `)}
`;