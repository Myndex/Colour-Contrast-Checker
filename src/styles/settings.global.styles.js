import { css, injectGlobal } from 'styled-components';
import { colors } from './settings.colors.styles';
import { typography } from './settings.typography.styles';
import { maxWidth } from './settings.breakpoints.styles';
import spacing from './settings.spacing.styles';

export default () => injectGlobal`
  :root {
    --background: ${colors.core};
    --foreground: ${colors.dark};
    --font: ${typography.family};
  }

  *,
  *:before,
  *:after {
    box-sizing: border-box;
    color: inherit;
    font-weight: normal;
  }

  html {
    font-size: 16px;
    -ms-text-size-adjust: 100%;
    -webkit-text-size-adjust: 100%;
  }

  body {
    margin: 0;
    background-color: var(--background);
    color: var(--foreground);
    font-family: var(--font);
    font-variation-settings: "wght" ${typography.weight.regular};
    line-height: ${typography.lineHeight.body};
    transition: background-color 0.3s ease-in-out;
    text-rendering: optimizeLegibility;
    -webkit-font-smoothing: antialiased;

    ${maxWidth('768', () => css`
      padding-bottom: 80px;
  `)}
  }

  body ::-moz-selection {
    background: var(--foreground);
    color: var(--background);
  }

  body ::selection {
    background: var(--foreground);
    color: var(--background);
  }

  h1,
  h2,
  p {
    margin-top: 0;
    margin-bottom: ${spacing.margin}px;
  }

  button {
    padding: 0;
    background-color: transparent;
    border: none;
    border-radius: 0;
    cursor: pointer;
    appearance: none;
  }

  input {
    margin: 0;
    padding: 0;
    background-color: transparent;
    border: none;
    border-radius: 0;
    appearance: none;
  }
`;
