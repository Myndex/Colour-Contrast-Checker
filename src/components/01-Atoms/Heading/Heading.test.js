import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-enzyme';
import 'jest-styled-components';
import { Heading1 } from './Heading.styles';
import { typography } from '../../../styles/settings.typography.styles';

it('Should render a heading 1 with tiny font size prop', () => {
  const component = <Heading1 tiny>Heading</Heading1>;
  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('font-size', typography.heading.size.tiny);
});

it('Should render a heading 1 with small font size prop', () => {
  const component = <Heading1 small>Heading</Heading1>;
  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('font-size', typography.heading.size.small);
});