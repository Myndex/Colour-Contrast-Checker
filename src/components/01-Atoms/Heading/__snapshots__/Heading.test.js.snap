// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a heading 1 with big font size prop 1`] = `
.c0 {
  font-family: Avenir Next,sans-serif;
  font-variation-settings: "wght" 600;
  line-height: 1.3;
  margin: 0 0 18px 0;
  font-size: calc(3em + 0.5vw);
}

<h1
  className="c0"
>
  Heading
</h1>
`;

exports[`Should render a heading 1 with large font size prop 1`] = `
.c0 {
  font-family: Avenir Next,sans-serif;
  font-variation-settings: "wght" 600;
  line-height: 1.3;
  margin: 0 0 18px 0;
  font-size: calc(4.5em + 0.5vw);
}

<h1
  className="c0"
>
  Heading
</h1>
`;

exports[`Should render a heading 1 with medium font size prop 1`] = `
.c0 {
  font-family: Avenir Next,sans-serif;
  font-variation-settings: "wght" 600;
  line-height: 1.3;
  margin: 0 0 18px 0;
  font-size: calc(2em + 0.5vw);
}

<h1
  className="c0"
>
  Heading
</h1>
`;

exports[`Should render a heading 1 with regular font size prop 1`] = `
.c0 {
  font-family: Avenir Next,sans-serif;
  font-variation-settings: "wght" 600;
  line-height: 1.3;
  margin: 0 0 18px 0;
  font-size: calc(1.375em + 0.5vw);
}

<h1
  className="c0"
>
  Heading
</h1>
`;

exports[`Should render a heading 1 with small font size prop 1`] = `
.c0 {
  font-family: Avenir Next,sans-serif;
  font-variation-settings: "wght" 600;
  line-height: 1.3;
  margin: 0 0 18px 0;
  font-size: calc(1em + 0.5vw);
}

<h1
  className="c0"
>
  Heading
</h1>
`;

exports[`Should render a heading 1 with tiny font size prop 1`] = `
.c0 {
  font-family: Avenir Next,sans-serif;
  font-variation-settings: "wght" 600;
  line-height: 1.3;
  margin: 0 0 18px 0;
  font-size: calc(0.625em + 0.5vw);
}

<h1
  className="c0"
>
  Heading
</h1>
`;

exports[`Should render a heading 1 with xl font size prop 1`] = `
.c0 {
  font-family: Avenir Next,sans-serif;
  font-variation-settings: "wght" 600;
  line-height: 1.3;
  margin: 0 0 18px 0;
  font-size: calc(7.75em + 0.5vw);
}

<h1
  className="c0"
>
  Heading
</h1>
`;
